xdem/biascorr.py

```diff
@@ -0,0 +1,830 @@
+        crs: rio.crs.CRS,  # Never None thanks to Coreg.fit() pre-process
+        weights: None | NDArrayf = None,
+        verbose: bool = False,
+        **kwargs,
+    ) -> None:
+
+        # Check number of variables
+        if len(bias_vars) != 2:
+            raise ValueError(
+                "Exactly two variables have to be provided through the argument 'bias_vars'"
+                ", got {}.".format(len(bias_vars))
+            )
+
+        super()._fit_func(
+            ref_dem=ref_dem,
+            tba_dem=tba_dem,
+            bias_vars=bias_vars,
+            transform=transform,
+            crs=crs,
+            weights=weights,
+            verbose=verbose,
+            **kwargs,
+        )
+
+
+class BiasCorrND(BiasCorr):
+    """
+    Bias-correction along N variables (e.g., simultaneously slope, curvature, aspect and elevation).
+    """
+
+    def __init__(
+        self,
+        fit_or_bin: Literal["bin_and_fit"] | Literal["fit"] | Literal["bin"] = "bin",
+        fit_func: Callable[..., NDArrayf]
+        | Literal["norder_polynomial"]
+        | Literal["nfreq_sumsin"] = "norder_polynomial",
+        fit_optimizer: Callable[..., tuple[NDArrayf, Any]] = scipy.optimize.curve_fit,
+        bin_sizes: int | dict[str, int | Iterable[float]] = 10,
+        bin_statistic: Callable[[NDArrayf], np.floating[Any]] = np.nanmedian,
+        bin_apply_method: Literal["linear"] | Literal["per_bin"] = "linear",
+    ):
+        """
+        Instantiate an N-D bias correction.
+
+        :param fit_or_bin: Whether to fit or bin. Use "fit" to correct by optimizing a function or
+            "bin" to correct with a statistic of central tendency in defined bins.
+        :param fit_func: Function to fit to the bias with variables later passed in .fit().
+        :param fit_optimizer: Optimizer to minimize the function.
+        :param bin_sizes: Size (if integer) or edges (if iterable) for binning variables later passed in .fit().
+        :param bin_statistic: Statistic of central tendency (e.g., mean) to apply during the binning.
+        :param bin_apply_method: Method to correct with the binned statistics, either "linear" to interpolate linearly
+            between bins, or "per_bin" to apply the statistic for each bin.
+        """
+        super().__init__(fit_or_bin, fit_func, fit_optimizer, bin_sizes, bin_statistic, bin_apply_method)
+
+    def _fit_func(  # type: ignore
+        self,
+        ref_dem: NDArrayf,
+        tba_dem: NDArrayf,
+        bias_vars: dict[str, NDArrayf],  # Never None thanks to BiasCorr.fit() pre-process
+        transform: rio.transform.Affine,  # Never None thanks to Coreg.fit() pre-process
+        crs: rio.crs.CRS,  # Never None thanks to Coreg.fit() pre-process
+        weights: None | NDArrayf = None,
+        verbose: bool = False,
+        **kwargs,
+    ) -> None:
+
+        # Check bias variable
+        if bias_vars is None or len(bias_vars) <= 2:
```

Comment:
This says less than or equal to 2, but the exception says more than two. Which one is correct?

Reply:
Fixed